companion/src/firmwares/edgetx/yaml_generalsettings.h

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright (C) OpenTX
+ *
+ * Based on code named
+ *   th9x - http://code.google.com/p/th9x
+ *   er9x - http://code.google.com/p/er9x
+ *   gruvin9x - http://code.google.com/p/gruvin9x
+ *
+ * License GPLv2: http://www.gnu.org/licenses/gpl-2.0.html
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "yaml_ops.h"
+#include "yaml_trainerdata.h"
+#include "yaml_calibdata.h"
+#include "yaml_switchconfig.h"
+
+#include "generalsettings.h"
+#include "eeprominterface.h"
+
+extern const YamlLookupTable beeperModeLut;
+extern const YamlLookupTable backlightModeLut;
+extern const YamlLookupTable bluetoothModeLut;
+extern const YamlLookupTable uartModeLut;
+extern const YamlLookupTable antennaModeLut;
+
+namespace YAML {
+
+  ENUM_CONVERTER(GeneralSettings::BeeperMode, beeperModeLut);
+
+  template<>
+  struct convert<GeneralSettings> {
+
+    // static YAML::Node encode(const GeneralSettings& rhs) {
+    //       Node node;
+    //       // TODO: convert struct into Node
+    //       return node;
+    // }
+
+    static bool decode(const Node& node, GeneralSettings& rhs) {
+
+      // TODO: check board string and fetch Board instance
+      std::string board;
+      node["board"] >> board;
+
+      // TODO: we should be able to fetch the board based on 'board'
+      auto fw = getCurrentFirmware();
+      if (fw && fw->getId().toStdString() != ("edgetx-" + board)) {
+        throw std::runtime_error("Wrong board");
+      }
```

Comment:
@elecpower instead of `getCurrentFirmware()` here, I think we should find the `Firmware*` that corresponds to the `board`, which is basically the `FLAVOUR` tag (examples: `tx16s`, `t18`, `x9d+`). Or do we even bother? Or just we just load based on the current profile selected, and it will be "converted" on the fly (loaded as if it would be such a radio).

Reply:
I would think, that we should just load it for the current profile. It is the users responsibility to select the right profile.

Reply:
Companion operates on the firmware linked to the profile. If the board and variant in the models and settings does not match the profile the models and settings are converted as much as is possible after loaded and any unsupported or iffy settings are highlighted to the user. There should be less need for conversions with yaml as that will be done during the parsing.

Reply:
Ok, so we probably don’t care at all, and will just ignore unsupported items (or flag them), based on the selected profile.